java-vtl-script/src/test/java/no/ssb/vtl/script/functions/VTLSubstrTest.java

```diff
@@ -0,0 +1,165 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+
+import no.ssb.vtl.model.VTLNumber;
+import no.ssb.vtl.model.VTLObject;
+import no.ssb.vtl.model.VTLString;
+import org.assertj.core.util.Lists;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+
+public class VTLSubstrTest implements VTLNumberFunctionTest {
+
+    private VTLSubstr vtlSubstr;
+
+    @Before
+    public void setUp() throws Exception {
+        vtlSubstr = VTLSubstr.getInstance();
+    }
+
+    @Test
+    @Override
+    public void testInvokeWithPositiveNumber() throws Exception {
+        VTLObject<?> result = vtlSubstr.invoke(
+                createArguments("Hello, world!", 2, 5)
+        );
+        assertThat(result).isNotNull();
+        assertThat(result).isEqualTo(VTLString.of("llo, "));
+
+        result = vtlSubstr.invoke(
+                createArguments("Hello, world!", 0, 4)
+        );
+        assertThat(result).isNotNull();
+        assertThat(result).isEqualTo(VTLString.of("Hell"));
+
+        result = vtlSubstr.invoke(
+                createArguments("", 0, 4)
+        );
+        assertThat(result).isNotNull();
+        assertThat(result).isEqualTo(VTLString.of(""));
+
+        //deviation from the VTL specification 1.1: return empty string
+        //if start position greater than the whole length of the input string
+        result = vtlSubstr.invoke(
+                createArguments("Hello", 5, 2)
+        );
+        assertThat(result).isNotNull();
+        assertThat(result).isEqualTo(VTLString.of(""));
+
+        //length argument will be ignored
+        result = vtlSubstr.invoke(
+                createArguments("Hello", 3, 3)
+        );
+        assertThat(result).isNotNull();
+        assertThat(result).isEqualTo(VTLString.of("lo"));
+    }
+
+
+    @Override
+    public void testInvokeWithNegativeNumber() throws Exception {
+        assertThatThrownBy(() -> vtlSubstr.invoke(
+                createArguments("Hello, world!", -1, 4)
+        ))
+                .as("exception when passing -1 as start position")
+                .hasMessage("Argument{name=startPosition, type=VTLInteger} must be greater than zero, was -1")
+                .isExactlyInstanceOf(IllegalArgumentException.class);
+
+        assertThatThrownBy(() -> vtlSubstr.invoke(
+                createArguments("Hello, world!", 1, -1)
+        ))
+                .as("exception when passing -1 as length")
+                .hasMessage("Argument{name=length, type=VTLInteger} must be greater than zero, was -1")
+                .isExactlyInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Override
+    public void testInvokeWithString() throws Exception {
+    }
```

Comment:
Did you forget to implement or to delete this test?

Reply:
The interface made me do this ;)